### DISCUSSION TITLE
Ability to Template Files

### DISCUSSION THREAD

**Discussion by dolfelt** [go-task/task#500] [Ideas]
We currently use an internal tool very similar to `go-task`. I'd like to replace it, however it has the ability to template files. It has a special command that allows files, `--vars` and `--varFiles` to be passed in, templated using Go templates (and sprig) and output to stdout.

This is especially useful for extending configuration files and providing a bit more power of Go templates and variables based on environments, parameters, etc.

Does anyone have thoughts on how to go about this or how it might specifically be implemented in `go-task`?

Thanks!

**Comment by dolfelt** [comment]
I found this tool which seems to pair nicely with what we are trying to do.

https://docs.gomplate.ca/

**Reply by andreynering** [reply]
I was a bit in doubt on how it worked without an example, but it's good you find a way to do what you need. 🙂

**Reply by dolfelt** [reply]
@andreynering Is there a way to print out all the variables to a yaml format I could pipe into this `gomplate` tool? That would greatly simplify the flow of data and commands. For example `echo "{{.YAML_VARS}}" | gomplate | kubectl -f -`

**Comment by ostcar** [comment]
I am also interested in creating files with task (wonderful tool by the way).

I did something similar then the example here: https://taskfile.dev/#/usage?id=gos-template-engine

It creates a file using cat. This works fine until a college from me run the task on windows.

Do you think it would be possible to add the possibility to create/replace files? For example
```yml
cmds:
- file: |
        content of file
```

**Reply by andreynering** [reply]
Regarding the lack of some shell builtins on Windows (including `cat`), there's a main issue here: https://github.com/go-task/task/issues/197.

**Comment by kundeng** [comment]
After so many years, still refusing to include a templating feature and ask user to resort to another tool.  Will never replace ansible is my guess.

**Reply by andreynering** [reply]
We don’t plan to replace Ansible, actually. Different tools.

Also, it’s not that we couldn’t consider a feature like this, but very few people asked for it.

If you’re interested, I suggest you to open an issue with this feature request, adding more details on how do you expect it to work.

**Comment by trulede** [comment]
@dolfelt I recently wrote a PR that uses the templating engine of Task to format the output of the --list CLI option. This was not a difficult thing to do, you can see the code here: https://github.com/go-task/task/pull/2348, and IIRC you had access to all the Task specific templating functions.

Perhaps its possible to have a command stereo type for `template` (in addition to `cmd` and `tasks`) where its possible to give the templaing engine a reference to the complied task (i.e. all the vars and envars), and from that generate a file.
